pynamodb/attributes.py

```diff
@@ -325,3 +337,233 @@ def deserialize(self, value):
         Takes a UTC datetime string and returns a datetime object
         """
         return parse(value, dayfirst=False).datetime
+
+
+class NullAttribute(Attribute):
+    attr_type = NULL
+
+    def serialize(self, value):
+        return True
+
+    def deserialize(self, value):
+        return None
+
+
+class MapAttributeMeta(type):
+    def __init__(cls, name, bases, attrs):
+        setattr(cls, '_attributes', None)
+
+
+class MapAttribute(with_metaclass(MapAttributeMeta, Attribute)):
+    attr_type = MAP
+
+    def __init__(self, hash_key=False, range_key=False, null=None, default=None, attr_name=None, **attrs):
+        super(MapAttribute, self).__init__(hash_key=hash_key,
+                                           range_key=range_key,
+                                           null=null,
+                                           default=default,
+                                           attr_name=attr_name)
+        self.attribute_values = {}
+        self._set_attributes(**attrs)
+
+    def __iter__(self):
+        return iter(self.attribute_values)
+
+    def __getitem__(self, item):
+        return self.attribute_values[item]
+
+    @classmethod
+    def _get_attributes(cls):
+        """
+        Returns the list of attributes for this class
+        """
+        if cls._attributes is None:
+            cls._attributes = AttributeDict()
+            for item in [attr for attr in dir(cls) if
```

Comment:
hmm, dir shouldn't really be used except as a help function. can you use getattr? and rewrite this longform? mixing list comps and for loops is confusing